pkg/steps/multi_stage/multi_stage.go

```diff
@@ -249,6 +273,30 @@ func (s *multiStageTestStep) Provides() api.ParameterMap {
 }
 func (s *multiStageTestStep) SubTests() []*junit.TestCase { return s.subTests }
 
+// getProfileData fetches the content of the cluster profile secret.
+// This is done both to guarantee it has been correctly imported into the test
+// namespace and to gather information used when generating the test pods.
+func (s *multiStageTestStep) getProfileData(ctx context.Context) error {
```

Comment:
Moved to its own function.

Reply:
I still think my suggestion makes more sense.
The goal of the caller of this function is to `readVPNData`, not `getProfileData`.
`getProfileData` is just a intermediate step.

But I am OK too if you keep it as it is.

Reply:
It's not just that, as the comment tries to explain.  We are also checking (somewhat implicitly now) that the profile secret has been created correctly, which is a requirement for the test.  I could not think of a better name, but it should reflect that this is not just about the VPN implementation (otherwise we'd conditionally call it in `run`).